packages/ketcher-core/src/domain/entities/monomer-chains/Chain.ts

```diff
@@ -49,7 +49,9 @@ export class Chain {
   public add(monomer: BaseMonomer) {
     this.createSubChainIfNeed(monomer);
 
-    if (monomer instanceof Sugar) {
+    if (monomer.monomerItem.props.unresolved) {
```

Comment:
I would suggest to use `monomer instanceof UnresolvedMonomer` where it is possible, instead of `monomer.monomerItem.props.unresolved`